src/main/java/edu/umich/its/cpm/BoxUtils.java

```diff
@@ -103,9 +164,100 @@ public static String getAuthCodeFromBoxCallback(HttpServletRequest request) {
 				break;
 			}
 		}
+		
+		if (authCode != null)
+		{
+			// now that we have the authCode, use it to get access token and fresh token
+		    // Returns a new access_token & refresh_token from an existing refresh_token
+		    // Each access_token is valid for 1 hour. In order to get a new, valid token, you can use the accompanying
+		    // refresh_token. Each refresh token is valid for 14 days. Every time you get a new access_token by using a
+		    // refresh_token, we reset your timer for the 14 day period. This means that as long as your users use your
+		    // application once every 14 days, their login is valid forever.
+		    // Args:
+		    //    - client_id: The client_id you obtained in the initial setup.
+		    //    - client_secret: The client_secret you obtained in the initial setup.
+		    //    - code: a string containing the code, or a dictionary containing the GET query
+		    // Returns:
+		    //    - a dictionary with the token and additional info
+			try
+			{
+				CloseableHttpClient httpclient = HttpClients.createDefault();
+				HttpPost httpPost = new HttpPost(boxTokenUrl);
+				List <NameValuePair> nvps = new ArrayList <NameValuePair>();
+				nvps.add(new BasicNameValuePair("grant_type", "authorization_code"));
+				nvps.add(new BasicNameValuePair("client_id", boxClientId));
+				nvps.add(new BasicNameValuePair("client_secret", boxClientSecret));
+				nvps.add(new BasicNameValuePair("code", authCode));
+				httpPost.setEntity(new UrlEncodedFormEntity(nvps));
+				CloseableHttpResponse response = httpclient.execute(httpPost);
+
+				try {
+					StatusLine statusLine = response.getStatusLine();
+				    log.info("token request status" + statusLine);
+				    if (statusLine.getStatusCode() == HttpStatus.SC_OK)
```

Comment:
Log the status code and message, and return null authcode, which will lead to further error log message as "Missing box integration parameters (Box client id, client secrect or authCode)"